driver/ixsystems/common.py

```diff
@@ -381,10 +381,21 @@ def _delete_snapshot(self, name, volume_name):
                 self.configuration.ixsystems_dataset_path + '/' + volume_name),
             name)
         LOG.debug('_delete_snapshot urn : %s', request_urn)
+        try:
+            ret = self.handle.invoke_command(FreeNASServer.SELECT_COMMAND,
+                                             request_urn, None)
+            LOG.debug('_delete_snapshot select response : %s', json.dumps(ret))
+            if ret['status'] == 'error' and ret['code'] == 404:
+                LOG.info("Attempting delete Cinder volume %s snapshot %s, however it cannot be found on TrueNAS"
+                         % (volume_name, name))
+                LOG.info("Assume TrueNAS admin delete it manually, proceeding with snapshot delete action on cinder side")
+                return
+        except Exception as e:
+            raise FreeNASApiError('Unexpected error', e)
```

Comment:
Before deleting snapshot check if it exists on TrueNAS, if already deleted, allows driver safely return without throw error to upper stream and cancel the whole snapshot delete action.  Note: 404 code is used by Truenas as resource not be found. If TrueNAS server is gone, general http connection exception will be thrown as FreeNASApiError cancel the whole transactions.